src/java.base/share/classes/jdk/internal/util/PlatformProps.java.template

```diff
@@ -0,0 +1,93 @@
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jdk.internal.util;
+
+import java.util.Locale;
+
+/**
+ * The corresponding source file is generated by GensrcMisc.gmk for java.base.
+ * @see OperatingSystem
+ * @see Architecture
+ */
+class PlatformProps {
+
+    // Index/ordinal of the current OperatingSystem enum as substituted by the build
+    static final OperatingSystem CURRENT_OS = initOS("@@OPENJDK_TARGET_OS@@");
+
+    // Precomputed booleans for each Operating System
+    static final boolean TARGET_OS_IS_LINUX   = "@@OPENJDK_TARGET_OS@@" == "linux";
+    static final boolean TARGET_OS_IS_MACOS   = "@@OPENJDK_TARGET_OS@@" == "macosx";
+    static final boolean TARGET_OS_IS_WINDOWS = "@@OPENJDK_TARGET_OS@@" == "windows";
+    static final boolean TARGET_OS_IS_AIX     = "@@OPENJDK_TARGET_OS@@" == "aix";
+
+    /**
+     * Returns the OperatingSystem of the build.
+     * Build time names are mapped to respective uppercase enum values.
+     * Names not recognized are throw an ExceptionInInitializerError with IllegalArgumentException.
+     */
+    private static OperatingSystem initOS(String osName) {
+        String mapped = switch (osName) {
+            case "macosx" -> "macos";
+            default -> osName;
+        };
+        return OperatingSystem.valueOf(mapped.toUpperCase(Locale.ROOT));
+    }
+
+    // The Architecture value for the current architecture
+    static final Architecture CURRENT_ARCH = initArch("@@OPENJDK_TARGET_CPU@@");
+
+    // Architecture.is64Bit() uses this value
+    static final int TARGET_ARCH_BITS = @@OPENJDK_TARGET_CPU_BITS@@;
+
+    // Architecture.isLittleEndian value from the build
+    static final boolean TARGET_ARCH_LITTLE_ENDIAN = "@@OPENJDK_TARGET_CPU_ENDIAN@@" == "little";
+
+    // Precomputed booleans for each Architecture, shared with jdk.internal.util.Architecture
+    // The variables are named to match the Architecture value names, and
+    // the values are named to match the build variables.
+    static final boolean TARGET_ARCH_IS_X64         = "@@OPENJDK_TARGET_CPU@@" == "x86_64";
```

Comment:
Would it be better to rename the enum entry to `X86_64`? I personally prefer the name x86-64 because I feel it is more regular than x64..

Reply:
> > Would be great if you could support "os.arch = ppc64" for AIX and legacy linux, too.
> 
> Changing os.arch is out of scope for this PR. The best way for that would someone supporting ppc to develop and propose a PR.

It seems that you replied in the wrong place.

Reply:
Likely, I misunderstood the request.  Mapping ppc64 to PPC64 is easy enough.
But currently, there is no direct support in the build/makefiles for ppc. See [OpenJDK supported platforms](https://wiki.openjdk.org/display/Build/Supported+Build+Platforms).
For anything other than the most trivial and obvious, someone who can build ppc and test should propose the changes.

Reply:
The naming of x86_64 was raised earlier and resolved to use X64.